sdk/nodejs/tests/runtime/langhost/cases/075.invoke_depends_on/index.js

```diff
@@ -0,0 +1,30 @@
+// Test the dependsOn invoke option.
+
+const assert = require("assert");
+const pulumi = require("../../../../../");
+
+const dependency = { resolved: false };
+
+class MyResource extends pulumi.CustomResource {
+    constructor(name) {
+        super("test:index:MyResource", name);
+    }
+}
+
+const dependsOn = pulumi.output(
+    new Promise((resolve) =>
+        setTimeout(() => {
+            dependency.resolved = true;
+            resolve(new MyResource("dependency"));
+        }),
+    ),
+);
+
+// By the time we serialise the arguments of the invoke, dependency.resolved
+// should be true due to dependsOn awaiting the setTimeout promise resolution
+// above. Without dependsOn, the invoke will be serialised before promise
+// resolution (since promises [microtasks] happen before timeout events
+// [macrotasks]).
+pulumi.runtime.invoke("test:index:echo", { dependency }, { dependsOn }).then((result) => {
```

Comment:
Doesn't this need to be `invokeOutput`?  I thought this is no longer supposed to work with plain invokes? Or what am I missing?

Reply:
You are right, this should use `invokeOutput`. This also reveals an issue that we might mistakenly pass `dependsOn` to `invoke` when not using TypeScript to intentionally bypassing the type checking.

I added a runtime check to `invoke` to guard against this and guide the user towards ${invoke}Output instead. I also added a test for this logging.